src/Doctrine/Orm/Filter/EndSearchFilter.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Doctrine\Orm\Filter;
+
+use ApiPlatform\Doctrine\Common\Filter\ManagerRegistryAwareInterface;
+use ApiPlatform\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Metadata\OpenApiParameterFilterInterface;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\Metadata\Parameter;
+use ApiPlatform\OpenApi\Model\Parameter as OpenApiParameter;
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\Persistence\ManagerRegistry;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+
+final class EndSearchFilter implements FilterInterface, ManagerRegistryAwareInterface, OpenApiParameterFilterInterface
+{
+    use FilterInterfaceTrait;
+
+    public function __construct(
+        private ?ManagerRegistry $managerRegistry = null,
+        private readonly ?array $properties = null,
+        private readonly ?NameConverterInterface $nameConverter = null,
+    ) {
+    }
+
+    protected function filterProperty(string $property, mixed $value, QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, ?Operation $operation = null, array $context = []): void
+    {
+        if (
+            null === $value
+            || !$this->isPropertyEnabled($property, $resourceClass)
+            || !$this->isPropertyMapped($property, $resourceClass, true)
+        ) {
+            return;
+        }
+
+        $alias = $queryBuilder->getRootAliases()[0];
+        $parameterName = $queryNameGenerator->generateParameterName($property);
+
+        $queryBuilder
+            ->andWhere(\sprintf('%s.%s LIKE :%s', $alias, $property, $parameterName))
+            ->setParameter($parameterName, '%'.$value);
+    }
+
+    public function getOpenApiParameters(Parameter $parameter): OpenApiParameter|array|null
+    {
+        return new OpenApiParameter(name: $parameter->getKey().'[]', in: 'query', style: 'deepObject', explode: true);
```

Comment:
I don't think we should hardcode parameter value style in the filter, I would rather be able to define the default strategy to use in the configuration and fallback to the default 
```suggestion
        return new OpenApiParameter(name: $parameter->getKey(), in: 'query');
```
should be suffisant for most of the cases.

Any thoughts @soyuka ?

Reply:
If you define your own OpenApiParameter this will not get called AFAIK

Reply:
Yes it's the final step (before supporting Odm) @soyuka @aegypius

Reply:
Will be done in a future PR (refer to the PRs in reference to this PR)